app/components/Avatar/Avatar.tsx

```diff
@@ -16,35 +15,31 @@ type Props = {
   className?: string;
 };
 
-@observer
-class Avatar extends React.Component<Props> {
-  @observable
-  error: boolean;
+function Avatar(props: Props) {
+  const { src, icon, showBorder, ...rest } = props;
 
-  static defaultProps = {
-    size: 24,
+  const [error, setError] = React.useState(false);
```

Comment:
Could potentially use `useBoolean` hook here to avoid defining `handleError` inline